build-tools/scripts/Jar.targets

```diff
@@ -7,6 +7,10 @@
       <_JavacSourceVersion Condition=" '$(_JavacSourceVersion)' == '' ">1.5</_JavacSourceVersion>
       <_JavacTargetVersion Condition="$(_JdkVersion.StartsWith ('9'))">1.8</_JavacTargetVersion>
       <_JavacTargetVersion Condition=" '$(_JavacTargetVersion)' == '' ">1.6</_JavacTargetVersion>
+      <JarPath Condition=" '$(JarPath)' == '' And !$([MSBuild]::IsOsPlatform(Windows)) ">$(JavaSdkDirectory)\bin\jar</JarPath>
+      <JarPath Condition=" '$(JarPath)' == '' And $([MSBuild]::IsOsPlatform(Windows)) ">$(JavaSdkDirectory)\bin\jar.exe</JarPath>
+      <JavaCPath Condition=" '$(JavaCPath)' == '' And !$([MSBuild]::IsOsPlatform(Windows)) ">$(JavaSdkDirectory)\bin\javac</JavaCPath>
+      <JavaCPath Condition=" '$(JavaCPath)' == '' And $([MSBuild]::IsOsPlatform(Windows)) ">$(JavaSdkDirectory)\bin\javac.exe</JavaCPath>
```

Comment:
I think you might be able to omit the `.exe` extensions here to make it simpler, and it would still work:
```
<JarPath Condition=" '$(JarPath)' == '' ">$(JavaSdkDirectory)\bin\jar</JarPath>
<JavaCPath Condition=" '$(JavaCPath)' == '' ">$(JavaSdkDirectory)\bin\javac</JavaCPath>
```
But then if I look at:
```
$ cat Configuration.OperatingSystem.props
<?xml version="1.0" encoding="utf-8"?>
<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <PropertyGroup>
        ...
        <JavaCPath Condition=" '$(JavaCPath)' == '' ">javac</JavaCPath>
        <JarPath Condition=" '$(JarPath)' == '' ">jar</JarPath>
        <JavaPath Condition=" '$(JavaPath)' == '' ">java</JavaPath>
    </PropertyGroup>
</Project>
```
Can the file that needs this, import `Configuration.props` instead?

Reply:
You're right, the `.exe` extension can be dropped. I'd rather not have to generate and rely on `Configuration.OperatingSystem.props` if possible though, and even if we made this change our tests would still pick up the first java tool(s) found in `$PATH` which might not be desirable. I'd rather rely on `<ResolveSdks/>` to set the path to various java tool(s) if their paths aren't explicitly set.